src/gt4py/next/embedded/nd_array_field.py

```diff
@@ -491,20 +508,50 @@ def _builtin_op(
             raise ValueError("Can only reduce local dimensions.")
         if axis not in field.domain.dims:
             raise ValueError(f"Field can not be reduced as it doesn't have dimension '{axis}'.")
+        if len([d for d in field.domain.dims if d.kind is common.DimensionKind.LOCAL]) > 1:
+            raise NotImplementedError(
+                "Reducing a field with more than one local dimension is not supported."
+            )
         reduce_dim_index = field.domain.dims.index(axis)
+        current_offset_provider = embedded_context.offset_provider.get(None)
+        assert current_offset_provider is not None
+        offset_definition = current_offset_provider[
+            axis.value
+        ]  # assumes offset and local dimension have same name
```

Comment:
Is this really the only way to do it? It feels so fragile...

Reply:
If we want to use the connecitivty as mask, I don't see any other way. That's why I want to clean up offset and dimensions...